### PR TITLE
refactor: docs(cci/network): readjust cci network resource support

### LINKED ISSUE #1722: [CCI] Add CCI network resource support
I'd like to add CCI network resource support.

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The current CCI network resource has the problems of parameter redundancy and poor ease of use.
For about these problems, resource should be readjusted:
- Deprecate redundant parameters.
- Refactor codes of errors caught.
- Refactor structure of acc test.
- Update related document.

Breaking changes:
- Update parameters naming:
  - Update from `available_zone` to `availability_zone`
  - Update from `security_group` to `security_group_id`
- Remove redundant parameters:
  - `domain_id`
  - `project_id`
- Update some parameter from arguments to attributes:
  - `subnet_id`
  - `cidr`

**Which issue this PR fixes**:
fixes #1722

**Special notes for your reviewer**:

**Release note**:

```release-note
1. refactor network schema parameters
2. support related document
3. update network acc test
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cci' TESTARGS='-run=TestAccCciNetwork_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cci -v -run=TestAccCciNetwork_basic -timeout 360m -parallel 4
=== RUN   TestAccCciNetwork_basic
=== PAUSE TestAccCciNetwork_basic
=== CONT  TestAccCciNetwork_basic
--- PASS: TestAccCciNetwork_basic (95.52s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cci       95.830s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
Pending golangsdk [PR](https://github.com/chnsz/golangsdk/pull/61) merge.

File: huaweicloud/services/cci/resource_huaweicloud_cci_network.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmtp.DiagErrorf("Error obtain HuaweiCloud CCI network status: %s", err)
+	}
+	d.SetId(create.Metadata.Name)
```
Comment (chengxiangdong) [inline huaweicloud/services/cci/resource_huaweicloud_cci_network.go]: after the resource is created, it is handed over to Terraform, so the `ID` should be set before waitfor

Comment (Lance52259) [reply huaweicloud/services/cci/resource_huaweicloud_cci_network.go]: The `Failed` state has been removed, and the ID is moved to the front of the waitfor method.

File: docs/resources/cci_network.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+variable "network_id" {}
+variable "security_group_id" {}
+
+data "huaweicloud_availability_zones" "test" {}
+
+resource "huaweicloud_cci_network" "test" {
+  availability_zone = data.huaweicloud_availability_zones.test.names[0]
+  namespace         = var.namespace_name
+  name              = var.network_name
+  network_id        = var.network_id
```
Comment (ShiChangkuo) [inline docs/resources/cci_network.md]: the `var.network_name` is confused with network_id

File: docs/resources/cci_network.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the CCI network.
+  If omitted, the provider-level region will be used. Changing this will create a new CCI network resource.
+
+* `availability_zone` - (Required, String, ForceNew) Specifies the availability zone (AZ) to which the CCI network
+  belongs. Changing this will create a new CCI network resource.
+
+* `namespace` - (Required, String, ForceNew) Specifies the namespace to logically divide your cloud container instances
+  into different group. Changing this will create a new CCI network resource.
+
+* `name` - (Required, String, ForceNew) Specifies the unique name of the PVC resource.
```
Comment (ShiChangkuo) [inline docs/resources/cci_network.md]: PVC ==> CCI network

File: huaweicloud/services/cci/resource_huaweicloud_cci_network.go [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+			Name:        d.Get("name").(string),
+			Annotations: resourceNetworkAnnotations(d, config),
+		},
+		Spec: networks.Spec{
+			AvailableZone: d.Get("availability_zone").(string),
+			NetworkType:   "underlay_neutron",
+			AttachedVPC:   subnet.VPC_ID,
+			NetworkID:     networkId,
+			SubnetID:      d.Get("subnet_id").(string),
+			Cidr:          d.Get("cidr").(string),
```
Comment (ShiChangkuo) [inline huaweicloud/services/cci/resource_huaweicloud_cci_network.go]: subnet_id and cidr are computed, they always be empty when creating

File: docs/resources/cci_network.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/cci_network.md]: Already use `vpc_network_id` instead of `network_id`.

File: docs/resources/cci_network.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/cci_network.md]: done

File: huaweicloud/services/cci/resource_huaweicloud_cci_network.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/cci/resource_huaweicloud_cci_network.go]: These arguments has been removed.

File: huaweicloud/resource_huaweicloud_compute_instance.go [hunk 5]
```diff
@@ -303,10 +303,9 @@ func ResourceComputeInstanceV2() *schema.Resource {
 				ForceNew: true,
 			},
 			"tags": {
-				Type:         schema.TypeMap,
-				Optional:     true,
-				ValidateFunc: utils.ValidateECSTagValue,
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_compute_instance.go]: please open another PR to fix the issue

**Pull request merged**